src/main/scala/cromwell/util/Encryption.scala

```diff
@@ -0,0 +1,75 @@
+package cromwell.util
+
+import java.security.SecureRandom
+import javax.crypto.Cipher
+import javax.crypto.spec.{IvParameterSpec, SecretKeySpec}
+
+import org.apache.commons.codec.binary.Base64
+
+import scala.util.{Failure, Success, Try}
+
+case object Aes256Cbc {
+
+  val encryption = "AES"
+  val blockSize = 128
+  val keySize = 256
+  val paddingMode = "PKCS5Padding"
+  val cipherMode = "CBC"
+
+  val ranGen = new SecureRandom()
+
+  final def init(mode: Int, secretKey: Array[Byte], iv: Array[Byte]) = {
+    val key = new SecretKeySpec(secretKey, encryption)
+    val ivSpec = new IvParameterSpec(iv)
+    val cipher = Cipher.getInstance(s"$encryption/$cipherMode/$paddingMode")
+    cipher.init(mode, key, ivSpec)
+    cipher
+  }
+
+  final def validateLength(arrayName: String, array: Array[Byte], expectedBitLength: Int): Try[Unit] = {
+    if (array.length * 8 == expectedBitLength) {
+      Success(())
+    } else {
+      Failure(new IllegalArgumentException(s"$arrayName size (${array.length * 8} bits) did not match the required length $expectedBitLength"))
+    }
+  }
+
+  implicit class validationChain(x: Try[Unit]) {
+    def validateAnotherLength(arrayName: String, array: Array[Byte], expectedBitLength: Int): Try[Unit] = {
```

Comment:
if only we had tools available to us which allowed us to accumulate successes & failures :)